File: app/src/main/java/com/airbnb/net/AirbnbService.java
package com.airbnb.net;

import com.airbnb.Message;

import java.util.List;

import retrofit.http.GET;
import rx.Observable;

interface AirbnbService {

    @GET("/search_tab")
    Observable<List<ListItem>> getSearchTabItems();

    @GET("/wish_list")
    Observable<List<HeroItem>> getWishList();

    @GET("/messages")
    Observable<List<Message>> getMessages();

}

File: app/src/main/java/com/airbnb/net/AirbnbService.kt
package com.airbnb.net

import com.airbnb.Message

import retrofit.http.GET
import retrofit.http.Path
import rx.Observable

interface AirbnbService {

    GET("/search_tab")
    public fun getSearchTabItems(): Observable<List<ListItem>>

    GET("/wish_list")
    public fun getWishList(): Observable<List<HeroItem>>

    GET("/messages")
    public fun getMessages(): Observable<List<Message>>

    GET("/listing/{id}")
    public fun getListing(Path("id") id: Long): Observable<Listing>

}
